src/shared/TempDirectory.cs

```diff
@@ -8,7 +8,7 @@
 
 namespace Microsoft.Tye
 {
-    internal class TempDirectory : IDisposable
+    public class TempDirectory : IDisposable
```

Comment:
😲

Reply:
this is already shared code? If we need it in other places I'm good with either making it public somewhere, or linking more places, but probably not both